File: testdata/fixtures/go/cookie-missing-httponly.go
package fixture

import (
	"net/http"
)

func vulnBare() *http.Cookie {
	// ruleid: cookie-missing-httponly
	return &http.Cookie{Name: "session", Value: "abc"}
}

func vulnPathOnly() *http.Cookie {
	// ruleid: cookie-missing-httponly
	return &http.Cookie{Name: "session", Value: "abc", Path: "/"}
}

func vulnSecureOnly() *http.Cookie {
	// ruleid: cookie-missing-httponly
	return &http.Cookie{Name: "session", Value: "abc", Secure: true}
}

func safeHttpOnly() *http.Cookie {
	// ok: cookie-missing-httponly
	return &http.Cookie{Name: "session", Value: "abc", HttpOnly: true}
}

func safeFull() *http.Cookie {
	// ok: cookie-missing-httponly
	return &http.Cookie{Name: "session", Value: "abc", Path: "/", Secure: true, HttpOnly: true}
}

func safeHttpOnlyFirst() *http.Cookie {
	// ok: cookie-missing-httponly
	return &http.Cookie{HttpOnly: true, Name: "session", Value: "abc"}
}

File: testdata/fixtures/java/cookie-missing-httponly.java
import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletResponse;

public class CookieHttpOnly {
    public void vuln(HttpServletResponse response) {
        Cookie c = new Cookie("session", "abc");
        // ruleid: cookie-missing-httponly
        response.addCookie(c);
    }

    public void vulnAgain(HttpServletResponse response) {
        Cookie c = new Cookie("session", "abc");
        // ruleid: cookie-missing-httponly
        response.addCookie(c);
    }

    public void vulnThird(HttpServletResponse response) {
        Cookie c = new Cookie("other", "xyz");
        // ruleid: cookie-missing-httponly
        response.addCookie(c);
    }

    public void safe(String name) {
        // ok: cookie-missing-httponly
        System.out.println(name);
    }

    public void plain() {
        // ok: cookie-missing-httponly
        System.out.println("addCookie in a comment is not a call");
    }

    public void nocookie(String name) {
        // ok: cookie-missing-httponly
        System.out.println(name);
    }
}
